libhoney.go

```diff
@@ -120,6 +121,38 @@ type Config struct {
 	Transport http.RoundTripper
 }
 
+// VerifyWriteKey calls out to the Honeycomb API to validate the write key, so
+// we can exit immediately if desired instead of happily sending events that
+// are all rejected.
+func VerifyWriteKey(config Config) error {
+	if config.WriteKey == "" {
+		return errors.New("Write key is empty")
+	}
+	url := fmt.Sprintf("%s/1/team_slug", config.APIHost)
+	req, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		return err
+	}
+	req.Header.Set("User-Agent", UserAgentAddition)
+	req.Header.Add("X-Honeycomb-Team", config.WriteKey)
+	client := &http.Client{}
+	resp, err := client.Do(req)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+	if resp.StatusCode == http.StatusUnauthorized {
+		return errors.New("Write key provided is invalid")
+	}
+	if resp.StatusCode != http.StatusOK {
+		body, _ := ioutil.ReadAll(resp.Body)
+		return errors.New(`Abnormal non-200 response verifying Honeycomb write key: %d
```

Comment:
This looks to be breaking the build, should be `fmt.Errorf`.

Reply:
oops :sweat_smile: good catch, will update